cmd/sync.go

```diff
@@ -0,0 +1,206 @@
+// Copyright © 2024 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/signal"
+
+	"github.com/ghodss/yaml"
+	"github.com/krayzpipes/cronticker/cronticker"
+	"github.com/spf13/cobra"
+
+	"github.com/bank-vaults/secret-sync/pkg/apis/v1alpha1"
+	"github.com/bank-vaults/secret-sync/pkg/provider"
+	"github.com/bank-vaults/secret-sync/pkg/storesync"
+)
+
+const (
+	flagSource   = "source"
+	flagTarget   = "target"
+	flagSyncJob  = "syncjob"
+	flagSchedule = "schedule"
+)
+
+type syncJob struct {
+	source *v1alpha1.StoreClient
+	target *v1alpha1.StoreClient
+	sync   *v1alpha1.SyncPlan
+}
+
+var syncCmd = &cobra.Command{
+	Use:   "sync",
+	Short: "Synchronizes secrets from a source to a target store based on sync strategy.",
+	RunE:  run,
+}
+
+func init() {
+	rootCmd.AddCommand(syncCmd)
+	syncCmd.PersistentFlags().StringP(flagSource, "s", "", "Source store config file. ")
+	_ = syncCmd.MarkPersistentFlagRequired(flagSource)
+	syncCmd.PersistentFlags().StringP(flagTarget, "t", "", "Target store config file. ")
+	_ = syncCmd.MarkPersistentFlagRequired(flagTarget)
+	syncCmd.PersistentFlags().String(flagSyncJob, "", "Sync job config file. ")
+	_ = syncCmd.MarkPersistentFlagRequired(flagSyncJob)
+
+	syncCmd.PersistentFlags().String(flagSchedule, "", "Sync periodically using CRON schedule. If not specified, runs only once.")
+}
```

Comment:
common approach is to expose command options in a form of a struct rather than using flag name consts

```suggestion
type syncJob struct {
	source *v1alpha1.StoreClient
	target *v1alpha1.StoreClient
	sync   *v1alpha1.SyncPlan
}

var syncCmd = &cobra.Command{
	Use:   "sync",
	Short: "Synchronizes secrets from a source to a target store based on sync strategy.",
	RunE:  run,
}

var syncCmdParams = struct {
	SourceStorePath string
	TargetStorePath string
	SyncJobPath     string
}{}

func init() {
	rootCmd.AddCommand(syncCmd)
	syncCmd.PersistentFlags().StringVarP(&syncCmdParams.SourceStorePath, "source", "s", "", "Source store config file. ")
	_ = syncCmd.MarkPersistentFlagRequired("source")
	// same for others
}
```

Reply:
I took the idea from Bank-Vaults CLI, but sure we can have a struct for params.

Reply:
The idea is from the `Bank-Vaults CLI`, I believe we should avoid working with pointers on our side, and offload this to `Cobra-CLI`, by having a SSOT for the params as constants.

Reply:
we could probably simplify more if we also opted in for similar approach for BV CLI

Reply:
Added the suggested change.